docs/components/nav_drawer.md

```diff
@@ -0,0 +1,323 @@
+<!--docs:
+title: "Material Navigation Drawer"
+layout: detail
+section: components
+excerpt: "Navigation drawers provide access to destinations in your app."
+iconId:
+path: /catalog/NavigationDrawers/
+-->
+
+# Navigation drawer
+
+A [navigation drawer](https://material.io/components/navigation-drawer) provides access to destinations in your app.
+
+![Hero navigation drawer image](assets/nav_drawer/NavDrawer-hero.png)
+
+## Contents
+
+- [Using a navigation drawer](#using-a-navigation-drawer)
+- [Anatomy](#anatomy)
+- [Types](#types)
+- [Standard navigation drawer](#standard-navigation-drawer)
+- [Modal navigation drawer](#modal-navigation-drawer)
+- [Bottom navigation drawer](#bottom-navigation-drawer)
+- [Theming a navigation drawer](#theming-a-navigation-drawer)
+
+## Using a navigation drawer
+
+A navigation drawer provides access to destinations and app functionality, such as switching accounts. It can either be permanently on-screen or controlled by a navigation menu icon.
+
+A navigation drawer is recommended for:
+
+- Apps with five or more top-level destinations
+- Apps with two or more levels of navigation hierarchy
+- Quick navigation between unrelated destinations
+
+Before you can use a Material navigation rail, you need to import the Material Components package for Flutter: `package:flutter/material.dart`.
+
+You need to use a `MaterialApp`.
+
+To create a modal drawer in Flutter, create a `Drawer` and set it to the `Scaffold`'s `drawer` property.
+
+The `child` of the drawer can be a `ListView` or a `Container` wrapping a `ListView`. The children of the `ListView` can consist of a `DrawerHeader` and `ListTile`s.
+
+There is no explicit bottom drawer in Flutter.
+
+To create a bottom drawer in Flutter, `showModalBottomSheet` can be used with a menu icon within a `BottomAppBar` in the `bottomNavigationBar` slot in the `Scaffold`.
+
+There is no explicit standard drawer in Flutter.
+
+To create a standard drawer in Flutter, create a `Row` with 2 children: a `Container` and a `Scaffold`, with the `Container` acting as the drawer.
+
+### Making navigation drawers accessible
+
+Flutter's drawer component APIs support labeling for accessibility via the `semanticLabel` property. The content within `child` can have additional accessibility labeling.
+
+For more guidance on writing labels, go to our page on how to write a good accessibility label.
+
+## Anatomy
+
```

Comment:
Include:

The following is a navigation drawer anatomy diagram:

Reply:
Done